Assets/MixedRealityToolkit/Inspectors/ServiceInspectors/ServiceFacadeInspector.cs

```diff
@@ -0,0 +1,405 @@
+        {
+            // Register this on startup so we can update whether a facade inspector is updated or not
+            SceneView.onSceneGUIDelegate += DrawSceneGUI;
+        }
+
+        private static List<string> dataProviderList = new List<string>();
+        private static Dictionary<Type, Type> inspectorTypeLookup = new Dictionary<Type, Type>();
+        private static Dictionary<Type, IMixedRealityServiceInspector> inspectorInstanceLookup = new Dictionary<Type, IMixedRealityServiceInspector>();
+        private static bool initializedServiceInspectorLookup = false;
+
+        Color proHeaderColor = (Color)new Color32(56, 56, 56, 255);
+        Color defaultHeaderColor = (Color)new Color32(194, 194, 194, 255);
+
+        const int headerXOffset = 48;
+        const int docLinkWidth = 175;
+
+        [SerializeField]
+        private Texture2D logoLightTheme = null;
+
+        [SerializeField]
+        private Texture2D logoDarkTheme = null;
+
+        protected virtual void Awake()
+        {
+            string assetPath = $"{MixedRealityEditorSettings.MixedRealityToolkit_RelativeFolderPath}/StandardAssets/Textures";
+
+            if (logoLightTheme == null)
+            {
+                logoLightTheme = (Texture2D)AssetDatabase.LoadAssetAtPath($"{assetPath}/MRTK_Logo_Black.png", typeof(Texture2D));
+            }
+
+            if (logoDarkTheme == null)
+            {
+                logoDarkTheme = (Texture2D)AssetDatabase.LoadAssetAtPath($"{assetPath}/MRTK_Logo_White.png", typeof(Texture2D));
+            }
+        }
+
+        protected override void OnHeaderGUI()
+        {
+            ServiceFacade facade = (ServiceFacade)target;
+            
+            // Draw a rect over the top of the existing header label
+            var labelRect = EditorGUILayout.GetControlRect(false, 0f);
+            labelRect.height = EditorGUIUtility.singleLineHeight;
+            labelRect.y -= labelRect.height;
+            labelRect.x = headerXOffset;
+            labelRect.xMax -= labelRect.x * 2f;
+
+            EditorGUI.DrawRect(labelRect, EditorGUIUtility.isProSkin ? proHeaderColor : defaultHeaderColor);
+
+            // Draw a new label
+            string header = facade.name;
+            if (string.IsNullOrEmpty(header))
+                header = target.ToString();
+
+            EditorGUI.LabelField(labelRect, header, EditorStyles.boldLabel);
+
+            /*GUILayout.BeginHorizontal();
+            GUILayout.FlexibleSpace();
+            GUILayout.Label(EditorGUIUtility.isProSkin ? logoDarkTheme : logoLightTheme, GUILayout.MaxHeight(128f));
+            GUILayout.FlexibleSpace();
+            GUILayout.EndHorizontal();*/
+        }
+
+        public override void OnInspectorGUI()
+        {
+            OnHeaderGUI();
+
+            ServiceFacade facade = (ServiceFacade)target;
+
+            if (facade.Service == null)
+            {   // Facade has likely been destroyed
+                return;
+            }
```

Comment:
It's interesting to see this check happen here but not in OnHeaderGUI - i.e. if the facade has been destroyed, is it still interesting to do the OnHeaderGUI stuff as well?

Reply:
Good question. I've never seen OnHeaderGUI get called for destroyed targets so Unity must do some under-the-hood check prior.